### PR TITLE
Update docker images and versions (most recent image - torch 2.9.1, cuda 13.0)

### PR DESCRIPTION
**Pull request by tyler-romero**
Most Images now support TransformerEngine and flash-attention-3
Adds support for cuda 13.0
Adds support for torch 2.9.1
Image used in CI is torch 2.9.1, cuda 12.8

---

> [!NOTE]
> Upgrades default/tooling to torch 2.9.1 with refreshed Docker/Beaker images (incl. CUDA 13.0), updates CI to new GPU image, and adapts fused CE loss to new API.
> 
> - **Build/Images**:
>   - Docker/Makefile: switch to Python `3.12`, Torch `2.9.1`; update `transformer-engine=2.9`, `liger-kernel=0.6.4`, `FLASH_ATTN_3_SHA`, add/pipe `FA3_MAX_JOBS` into build; keep CUDA `12.8.1` base.
>   - Beaker images: default Torch -> `2.9.1`; refresh stable tags and namespaces; add `stable_cu130` (CUDA 13.0); list images for `tch280`, `tch271`, etc.
> - **CI**:
>   - `.github/workflows/docker.yml`: new runner, Python `3.12`, matrix CUDA `12.8.1/12.9.1`, Torch `2.7.1/2.9.0`, pass `FA3_MAX_JOBS`; disable push trigger.
>   - `.github/workflows/main.yml`: GPU tests use `tylerr/olmo-core-tch291cu128-2025-11-25` via anchor.
> - **Core**:
>   - `cross_entropy_loss`: handle extra `per_token_acc` return from liger fused op and discard it.
> - **Scripts**:
>   - Training scripts switch Beaker image to `OLMoCoreBeakerImage.tch271_cu126`.
> - **Docs**:
>   - `CHANGELOG.md`: note stable Torch `2.9.1` and updated libs in Beaker images.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a339bafc007a35bbbdd5ea54f7a00a09d52ad917. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -16,16 +19,16 @@ on:
...
+  #   #   - main
+  #   tags:
+  #     - 'v*.*.*'
 
 jobs:
   docker:
-    name: CUDA ${{ matrix.cuda }} ${{ matrix.target }}
-    runs-on: ubuntu-latest-m
+    name: CUDA ${{ matrix.cuda }} Torch ${{ matrix.torch.version }}
+    runs-on: 8-Core-XL-Runner-Ubuntu-Latest  # 8 cores, 32GB memory
```
Comment (epwalsh) [inline .github/workflows/docker.yml:31]: Do we still OOM on this type of runner?

**Review by epwalsh — APPROVED** [review]
Nice!

File: .github/workflows/docker.yml (hunk 1 above)
Comment (tyler-romero) [reply .github/workflows/docker.yml:31]: Yes sadly, when compiling fa3, even if MAX_JOBS is set to something really low like 4 (which also makes it take forever)
